client/src/services/data.js

```diff
@@ -0,0 +1,52 @@
+import {logDebug} from './log';
+
+const states = {};
+
+/**
+ * Fetch the list of code review.
+ *
+ * @param {string} channel
+ * @param {number} limit
+ * @param {number} page
+ * @param {string} status
+ * @returns {Promise<T|null>}
+ */
+export const fetchReviews = async ({channel, limit, page, status}) => {
+    const key = `fetchReviews-${channel}-${limit}-${page}-${status}`;
+
+    if (states[key]) {
+        return null;
```

Comment:
The side effects in this function don't feel right - I would keep them out and let this function just do the job of fetching and have another function manage state.

Reply:
I refactor and align the code with React 18.  There is issue cause multiple fetch and rendering is due to dev mode :( 
did some google search turn on a few articles about it, eg. https://levelup.gitconnected.com/react-18-the-trickiness-of-useeffect-fadfa65fa4b4